checkstyle/import-control.xml

```diff
@@ -99,6 +100,7 @@
       <allow pkg="org.apache.kafka.common.config" />
       <allow pkg="org.apache.kafka.common.metrics" />
       <allow pkg="org.apache.kafka.common.security" />
+      <allow class="org.apache.kafka.common.requests.ApiVersionsResponse" />
```

Comment:
Hmm.. Do you have any ideas? I guess I would say that the dependency is there regardless of how we choose to hide it since we have to go through `ChannelBuilders` to build the authenticator, and the authenticator does depend on the request APIs.

Reply:
Yeah, the API versions response is pretty special in the protocol. Fortunately or unfortunately, it's not treated as a generic response.  I tried to think of some ways to avoid this dependency but they all ended up being kind of like obfuscation.

Even if we chose to make this a `Supplier<AbstractResponse>`, that's still in the requests package....

Reply:
I meant that you could introduce an interface that the response implements. That interface would live in the network package.

Reply:
Hmm.  Let's revisit this after 2.8.